apps/prairielearn/elements/pl-xss-safe/pl-xss-safe.mustache

```diff
@@ -15,7 +15,12 @@
     code = DOMPurify.sanitize(code);
     const shadowRoot = element.attachShadow({ mode: 'open' });
```

Comment:
Is there a reason this element doesn't do the same

```
  let shadowRoot = preview.shadowRoot;
  if (!shadowRoot) {
  ```
  
  logic?

Reply:
Because this code is only running once at initialization. The pl-file-editor code runs every time the content changes.

Reply:
Note that `attachShadow` returns the existing shadow root if for some reason it has already been attached.